modules/report/includes/commerce_pos_report.pages.inc

```diff
@@ -391,33 +391,29 @@ function commerce_pos_report_get_totals($date_filter, $register_id) {
 
   // Populate the arrays with the POS Payment methods.
   $pos_methods = commerce_pos_get_payment_methods();
-  $currencies = commerce_currencies(TRUE);
 
+  // Default to a single currency based off of the default commerce currency.
+  $currency_code = commerce_default_currency();
+
+  // Build a list of default totals.
   foreach ($pos_methods as $pos_method) {
-    if (!empty($pos_method['status']) && intval($pos_method['status']) === COMMERCE_POS_PAYMENT_METHOD_STATUS_ENABLED) {
-      foreach (array_keys($currencies) as $currency_code) {
-        $totals[$currency_code][$pos_method['id']] = array(
-          CommercePosService::TRANSACTION_TYPE_SALE => 0,
-          CommercePosService::TRANSACTION_TYPE_RETURN => 0,
-          CommercePosService::TRANSACTION_TYPE_EXCHANGE => 0,
-        );
-      }
-      $transaction_summary[$pos_method['id']] = array();
-      $transaction_counts[$pos_method['id']] = 0;
+    if (empty($pos_method['status']) || intval($pos_method['status']) === COMMERCE_POS_PAYMENT_METHOD_STATUS_ENABLED) {
+      $totals[$currency_code][$pos_method['id']] = array(
+        CommercePosService::TRANSACTION_TYPE_SALE => 0,
+        CommercePosService::TRANSACTION_TYPE_RETURN => 0,
+        CommercePosService::TRANSACTION_TYPE_EXCHANGE => 0,
+      );
     }
+    $transaction_summary[$pos_method['id']] = array();
+    $transaction_counts[$pos_method['id']] = 0;
   }
 
   $visible_statuses = variable_get('commerce_pos_receipt_visible_payment_status', array(COMMERCE_PAYMENT_STATUS_SUCCESS, COMMERCE_PAYMENT_STATUS_VOID));
   $statuses = commerce_payment_transaction_statuses();
 
   foreach ($query->execute() as $row) {
-
     if (!isset($totals[$row->commerce_order_total_currency_code][$row->payment_method])) {
-      $totals[$row->commerce_order_total_currency_code][$row->payment_method] = array(
-        CommercePosService::TRANSACTION_TYPE_SALE => 0,
-        CommercePosService::TRANSACTION_TYPE_RETURN => 0,
-        CommercePosService::TRANSACTION_TYPE_EXCHANGE => 0,
-      );
+      continue;
```

Comment:
The reason we continue here is so we don't add extra currencies or payment methods that shouldn't be here. I just wonder if we should watchdog or set a message here somehow.